sycl/include/sycl/known_identity.hpp

```diff
@@ -63,14 +63,24 @@ using IsLogicalOR =
     bool_constant<std::is_same<BinaryOperation, sycl::logical_or<T>>::value ||
                   std::is_same<BinaryOperation, sycl::logical_or<void>>::value>;
 
+template <typename T>
+using isComplex =
+    bool_constant<std::is_same<T, std::complex<short>>::value ||
+                  std::is_same<T, std::complex<int>>::value ||
```

Comment:
Why do we have these two instantiations of `std::complex<>` for integral `T`?

I've only ever seen `std::complex<float>`, `std::complex<double>` and `std::complex<long double>`.  Both  [`sycl_ext_oneapi_complex`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_complex.asciidoc) and [`sycl_ext_oneapi_complex_algorithms`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_algorithms.asciidoc) only support `float`, `double` and `sycl::half`.

Reply:
Thanks for bringing this up. I wasn't aware that for std::complex, we only support float and double types. I've accordingly made the changes.